_posts/en/newsletters/2025-01-31-newsletter.md

```diff
@@ -0,0 +1,284 @@
+  fixed by "upgrading to LDK version 0.1 and replaying the sequence of
+  commitment and HTLC transactions that led to the lock up."
+
+  However, the release candidate LDK 0.1-beta changed its logic (see
+  [Newsletter #335][news335 ldk3340]) and began batching all types of
+  HTLCs together, which could allow an attacker to create a conflict
+  with a timelocked HTLC.  If the resolution of that HTLC remained stuck
+  after the timelock expired, theft was possible.  Upgrading to the
+  release version of LDK 0.1 also fixes this form of the vulnerability.
+
+  Morehouse's post provides additional details and discusses possible
+  ways to prevent future vulnerabilities stemming from the same root
+  cause.
+
+- **Replacement cycling attacks with miner exploitation:** Antoine Riard
+  [posted][riard minecycle] to the Bitcoin-Dev mailing list to
+  disclose an additional vulnerability possible with the [replacement
+  cycling][topic replacement cycling] attack he originally publicly
+  disclosed in 2023 (see [Newsletter #274][news274 cycle]).  In short:
+
+  1. Bob broadcasts a transaction paying Mallory (and possibly other
+     people).
+
+  2. Mallory [pins][topic transaction pinning] Bob's transaction.
+
+  3. Bob doesn't realize he's been pinned and bumps the fees (using
+     either [RBF][topic rbf] or [CPFP][topic cpfp]).
+
+  4. Because Bob's original transaction was pinned, his fee bump
+     doesn't propagate.  However, Mallory receives it somehow.  Steps 3
+     and 4 may repeat multiple times to
+     greatly increase Bob's fee.
+
+  5. Mallory mines Bob's highest fee bump, which nobody else is trying
+     to mine because it didn't propagate.  This allows Mallory to earn
+     more fees than other miners.
+
+  6. Mallory can now use replacement cycling to move her transaction pin
+     to another transaction and repeat the attack (possibly with a
+     different victim) without allocating additional funds, making the
+     attack economically efficient.
+
+  We don't judge the vulnerability to be a significant risk.
+  Exploiting the vulnerability requires specific circumstances that
+  might be rare and can result in an attacker losing money if they judge
+  network conditions poorly.  If an attacker did regularly
+  exploit the vulnerability, we believe their behavior would be detected
+  by community members who build and use [block monitoring
+  tools][miningpool.observer].
+
+- **Updated stats on compact block reconstruction:** following up on a
+  previous thread (see [Newsletter #315][news315 cb]), developer 0xB10C
+  [posted][b10c cb] to Delving Bitcoin updated stats on the frequency at
+  which his Bitcoin Core nodes needed to request additional transactions
+  to perform [compact block][topic compact block relay]
+  reconstruction.  When a nod ereceives a compact block, it must request
+  any transactions in that block that they it does not already have in its
```

Comment:
```suggestion
  any transactions in that block that it does not already have in its
```

Also “which” instead of “that”? Unsure.

```suggestion
  any transactions in that block, which it does not already have in its
```